specs/enable-selinux.adoc

```diff
@@ -0,0 +1,197 @@
+to be attacked if it does not managed by Selinux completely. A normal user
+with no specific privileges on the system who is trying to interact with one of
+the root-running processes that can suddenly misbehave and give the user
+root access or allows the user to launch root access commands.
+
+Currently tendrl services like gluster-integration, node-agent, api,
+monitoring-integration services are running as unconfined services.
+ex1: system_u:system_r:unconfined_service_t:s0 18240 ? 02:51:40 tendrl-node-age
+ex2: system_u:system_r:unconfined_service_t:s0 18240 ? 02:51:40 tendrl-api
+Unconfined Service issue:- This will leads the 'privileges of the process'
+being attacked easily. Tendrl process that run as root are prone to be attacked
+to get root access on the system.
+
+Tendrl services does not belong to any selinux domain. Proper selinux domain should
+be assigned to every tendrl services to identify what is allowed for this service.
+
+Currently tendrl does not have enough confined rules for files being used by
+its process. Contexts for files used by tendrl should also be specified clearly.
+So that the resource can be used with the restricted gated privilege.
+The file or directory created in a directory should also acquire same context.
+
+When the system runs SELinux in permissive mode, users are able to
+label files incorrectly. Files created with SELinux in permissive mode are not
+labeled correctly while files created while SELinux is disabled are not labeled
+at all. This behavior causes problems when changing to enforcing mode.
+Also there are some application that know about SELinux status can change their
+behavior when selinux is in permissive mode.
+
+== Use Cases
+
+* Enable SELinux security system for tendrl
+* Retain the existing security measures in an already enabled SELinux system
+
+== Proposed change
+
+* Running a tendrl related services in a Specific Security Context
+  Tendrl can have a selinux policies for all tendrl related services
+  Each tendrl proces can be assigned to an unique security context which,
+  just like with the user under which the process runs, helps Linux in
+  identifying what the application should and shouldn't be allowed to do.
+
+* Marking only the required type as permissive
+  In addition to have SELinux policies for tendrl, it would better to mark
+  only the required type as permissive initially; instead of making the
+  entire system permissive, tendrl could make only the purticular domain as permissive.
+  Over the release iterations, testing various test cases, SELinux can be
+  compleatly enforced. Because it's inevitable to completely block any new
+  avc errors which might come on any new updates to tendrl.
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+None
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+SELinux policy files will be added in to this tendrl-api module.
+This set of policies will be used at tendrl server.
```

Comment:
I hope ceph and gluster has its own selinux policy. 
it would be a heavy work if we need to write policy for that also.
However, if required we can add policy for that service also.